Source/WebCore/rendering/RenderBlockFlow.cpp

```diff
@@ -1367,6 +1394,55 @@ bool RenderBlockFlow::shouldTrimChildMargin(MarginTrimType marginTrimType, const
     return lastInFlowChildBox() == &child;
 }
 
+void RenderBlockFlow::trimFloatBlockEndMargins(LayoutUnit blockFormattingContextInFlowContentHeight)
+{
+    ASSERT(m_floatingObjects && establishesBlockFormattingContext());
+    auto computeFloatMarginAfterStartLocationInBlockFormattingContext = [&](const RenderBox& floatBox) {
+        // Start with the location of the float within its containing block. If this containing
+        // block is the BFC in which we are trimming, then we can use this as the starting location
+        // since this is the float's positin within the BFC. If not, add the containing block's location
+        // to the float's location. This new value should be the location of the float relative to this
+        // new containing block. Keep doing this until we find the containing block that establishes
+        // a BFC
+        auto bfcRoot = floatBox.blockFormattingContextRoot();
+        ASSERT(bfcRoot);
+
+        auto startLocation = floatBox.localToAbsolute();
+        auto containingBlock = floatBox.containingBlock();
+        while (!containingBlock->establishesBlockFormattingContext()) {
+            startLocation += containingBlock->localToAbsolute();
+            containingBlock = containingBlock->containingBlock();
+        }
```

Comment:
I am bit puzzled about accumulating absolute coordinates here. To my understanding localToAbsolute() returns the box's absolute coordinate, relative to RenderView.

Reply:
I stepped through the calls and it seems like `RenderBox::mapLocalToContainer` gets nullptr for `ancestorContainer` and `view().frameView().layoutContext().isPaintOffsetCacheEnabled()` is true. In that case it looks like it uses `locationOffset` and returns early. Later parts of the code seem to call up the containers, though. 

I'm not super familiar with the intricacies of this function, so definitely might not be the right thing to use. Maybe instead we could use `offsetFromContainer` until we get up to the BFC root instead? Thoughts?

Reply:
Changed to use `offsetFromContainer` instead